libs/wire-api/src/Wire/API/Provider/Service.hs

```diff
@@ -126,37 +124,44 @@ data ServiceKey = ServiceKey
   }
   deriving stock (Eq, Show, Generic)
   deriving (Arbitrary) via (GenericUniform ServiceKey)
+  deriving (FromJSON, ToJSON, S.ToSchema) via Schema ServiceKey
 
-instance ToJSON ServiceKey where
-  toJSON k =
-    object
-      [ "type" .= serviceKeyType k,
-        "size" .= serviceKeySize k,
-        "pem" .= serviceKeyPEM k
-      ]
-
-instance FromJSON ServiceKey where
-  parseJSON = withObject "ServiceKey" $ \o ->
-    ServiceKey
-      <$> o .: "type"
-      <*> o .: "size"
-      <*> o .: "pem"
+instance ToSchema ServiceKey where
+  schema =
+    object "ServiceKey" $
+      ServiceKey
+        <$> serviceKeyType .= field "type" schema
+        <*> serviceKeySize .= field "size" schema
+        <*> serviceKeyPEM .= field "pem" schema
 
 -- | Other types may be supported in the future.
 data ServiceKeyType
   = RsaServiceKey
   deriving stock (Eq, Enum, Bounded, Show, Generic)
   deriving (Arbitrary) via (GenericUniform ServiceKeyType)
+  deriving (FromJSON, ToJSON, S.ToSchema) via Schema ServiceKeyType
 
-instance ToJSON ServiceKeyType where
-  toJSON RsaServiceKey = String "rsa"
-
-instance FromJSON ServiceKeyType where
-  parseJSON (String "rsa") = pure RsaServiceKey
-  parseJSON _ = fail "Invalid service key type. Expected string 'rsa'."
+instance ToSchema ServiceKeyType where
+  schema =
+    enum @Text "ServiceKeyType" (element "rsa" RsaServiceKey)
+      & S.schema . S.example ?~ pem
+    where
+      pem =
+        A.String . Text.unlines $
+          [ "-----BEGIN PUBLIC KEY-----",
+            "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAu+Kg/PHHU3atXrUbKnw0",
+            "G06FliXcNt3lMwl2os5twEDcPPFw/feGiAKymxp+7JqZDrseS5D9THGrW+OQRIPH",
+            "WvUBdiLfGrZqJO223DB6D8K2Su/odmnjZJ2z23rhXoEArTplu+Dg9K+c2LVeXTKV",
+            "VPOaOzgtAB21XKRiQ4ermqgi3/njr03rXyq/qNkuNd6tNcg+HAfGxfGvvCSYBfiS",
+            "bUKr/BeArYRcjzr/h5m1In6fG/if9GEI6m8dxHT9JbY53wiksowy6ajCuqskIFg8",
+            "7X883H+LA/d6X5CTiPv1VMxXdBUiGPuC9IT/6CNQ1/LFt0P37ax58+LGYlaFo7la",
+            "nQIDAQAB",
+            "-----END PUBLIC KEY-----"
+          ]
```

Comment:
This example seems wrong, aren't we just parsing the literal string `"rsa"` here? Maybe you wanted to put this in the instance for `ServiceKeyPEM`?

Reply:
Yes, it was supposed to go with `ServiceKeyPEM`, nice catch!